dwave/system/samplers/clique.py

```diff
@@ -0,0 +1,186 @@
+        parameters.pop('quadratic_range', None)
+
+        return parameters
+
+    @property
+    def properties(self):
+        try:
+            return self._properties
+        except AttributeError:
+            pass
+
+        self._properties = dict(qpu_properties=self.child.properties)
+        return self.properties
+
+    @property
+    def largest_clique_size(self):
+        """The maximum number of variables."""
+        return len(self.largest_clique())
+
+    def largest_clique(self):
+        """Return a largest-size clique embedding."""
+        return busgraph_cache(self.target_graph).largest_clique()
+
+    def sample(self, bqm, chain_strength=None, **kwargs):
+        """Sample from the specified binary quadratic model.
+
+        Args:
+            bqm (:class:`~dimod.BinaryQuadraticModel`):
+                Any binary quadratic model with up to
+                :attr:`.largest_clique_size` variables. This BQM is embedded
+                using a dense clique embedding.
+
+            chain_strength (float, optional):
+                The (relative) chain strength to use in the embedding. By
+                default a chain strength of `1.5sqrt(N)` where `N` is the size
+                of the largest clique, as returned by
+                :attr:`.largest_clique_size`.
+
+            **kwargs:
+                Optional keyword arguments for the sampling method, specified
+                per solver in :attr:`.DWaveCliqueSampler.parameters`.
+                D-Wave System Documentation's
+                `solver guide <https://docs.dwavesys.com/docs/latest/doc_solver_ref.html>`_
+                describes the parameters and properties supported on the D-Wave
+                system. Note that `auto_scale` is not supported by this
+                sampler, because it scales the problem as part of the embedding
+                process.
+
+        """
+
+        # some arguments should not be overwritten
+        if 'auto_scale' in kwargs:
+            raise TypeError("sample() got an unexpected keyword argument "
+                            "'auto_scale'")
+        if 'bias_range' in kwargs:
+            raise TypeError("sample() got an unexpected keyword argument "
+                            "'bias_range'")
+        if 'quadratic_range' in kwargs:
+            raise TypeError("sample() got an unexpected keyword argument "
+                            "'quadratic_range'")
+
+        # handle circular import. todo: fix
+        from dwave.system.composites.embedding import FixedEmbeddingComposite
+
+        # get the embedding
+        embedding = find_clique_embedding(bqm.variables, self.target_graph,
+                                          use_cache=True)
+
+        if chain_strength is None:
+            # chain length determines chain strength
+            if embedding:
+                chain_length = len(next(iter(embedding.values())))
+                chain_strength = 1.5 * math.sqrt(chain_length)
```

Comment:
A few issues

1. It assumes constant chain length by only looking at the first embedded variable. But this is not important given point 2.
2. The N in 1.5*sqrt(N) should be the size of the clique, not the chain length
3. The 1.5 prefactor works for Chimera, and I think for Pegasus needs to be more like 1.3. Worth doublechecking with @jackraymond

Reply:
Good catch, thanks

Reply:
I have fixed `N`. For the 1.3 vs 1.5, I am inclined to go with 1.5 for now, as a vast improvement over the previous value. We can adjust performance in a subsequent PR.

Reply:
Whether we use 1.5 or 1.3 is mute if we do not scale to the logical problem.
Define logical problem as 
H = sum_{i<j } Jij s_i s_j + sum_i h_i s_i (matches BQM definition). 
Define 
RMS(J) = [sum_{i<j} J_ij^2]/[N(N-1)/2].  Note that num_edges= N(N-1)/2. 
We should define chain strength as 
CS = A RMS(J) sqrt(N) 

A = 1.5 is a good choice for spin glasses on Bay2 (DW2000Q)
A = 1.2 is a better choice for Bay17 (Advantage)
Either choice is much better than the current default.

Reply:
Moving the RMS to a subsequent PR, see https://github.com/dwavesystems/dwave-system/issues/318